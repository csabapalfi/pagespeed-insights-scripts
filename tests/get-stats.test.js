const {getStats} = require('../lib/get-stats');

describe('stats', () => {
  const mockSample = (data) => ({
    type: 'result', name: 'run 1', ...data
  });

  const samples = [
    mockSample({score: 80, FCP: 1, FMP: 2, SI: 2, FCI: 5, TTI: 6}),
    mockSample({score: 90, FCP: 2, FMP: 3, SI: 3, FCI: 6, TTI: 7}),
  ]

  it('returns expected stats', () => {
    expect(getStats(samples))
      .toMatchSnapshot();
  });
});