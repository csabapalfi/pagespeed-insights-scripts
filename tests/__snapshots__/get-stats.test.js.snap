// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats returns expected stats 1`] = `
Array [
  Object {
    "FCI": 5.5,
    "FCP": 1.5,
    "FMP": 2.5,
    "SI": 2.5,
    "TTI": 6.5,
    "fetchTime": NaN,
    "name": "median",
    "score": 85,
    "type": "statistic",
  },
  Object {
    "FCI": 0.7071067811865476,
    "FCP": 0.7071067811865476,
    "FMP": 0.7071067811865476,
    "SI": 0.7071067811865476,
    "TTI": 0.7071067811865476,
    "fetchTime": NaN,
    "name": "stddev",
    "score": 7.0710678118654755,
    "type": "statistic",
  },
  Object {
    "FCI": 5,
    "FCP": 1,
    "FMP": 2,
    "SI": 2,
    "TTI": 6,
    "fetchTime": "21:48:05",
    "name": "min",
    "score": 80,
    "type": "statistic",
  },
  Object {
    "FCI": 6,
    "FCP": 2,
    "FMP": 3,
    "SI": 3,
    "TTI": 7,
    "fetchTime": "21:48:05",
    "name": "max",
    "score": 90,
    "type": "statistic",
  },
]
`;
